flang/include/flang/Evaluate/constant.h

```diff
@@ -141,6 +141,10 @@ template <typename T> class Constant : public ConstantBase<T> {
     }
   }
 
+  std::vector<Scalar<T>> GetArrayValue() const {
+    return Base::values_;
+  }
```

Comment:
@jeanPerier  - Is this presently missing functionality?

Reply:
Yes, on purpose. In semantics Peter did not wanted people to access the raw array and make their own multidimensional indexing. Using `At` + constant subscripts is safe way to interact with it.
Now, here, if the array constant lowering code  is always using a 1d loop that's fine to add this accessor, but maybe the general case would be better using `At()` if it needs to do index calculation.

Reply:
When I work on multi dimensional arrays, I shall switch to the alternative method that you suggest here. It's likely that work would subsume even the 1D array case.

Reply:
It seems like a good idea to explore the `At` interface now. It's OK to defer the implementation of multi-dimensional arrays, but we need to grok the interfaces in the front-end.

Reply:
Done.